sound/soc/sof/card-tplg-ops.c

```diff
@@ -0,0 +1,120 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause)
```

Comment:
@bardliao an alternative thought. Can we move this to the machine driver? I think it makes sense for this code to coexist along with the machine info and dailinks there.

Reply:
I thought about it. However, the topology stuff is more like belong to component level instead of card level. Besides, we need the `*sof_pdata` data to construct the tplg file name.

Reply:
you've got tplg file name is available also in struct snd_soc_acpi_mach. I dont think you really need sof_pdata for this

Reply:
> I thought about it. However, the topology stuff is more like belong to component level instead of card level. Besides, we need the `*sof_pdata` data to construct the tplg file name.

I agree with topology stuff belonging to component but this has nothing to do with the topology stuff but rather the machine level stuff ie whether there's a mic, a speaker a dmic etc no?

Reply:
@ranj063 I moved the ops to ` sound/soc/intel/common/`, and set the ops in each acpi_mach that needs it. I don't set the ops in sof_sdw.c because I can't access the same acpi_mach pointer in sof_sdw.c and SOF driver. The acpi_mach data is copied to the platform device not the pointer.